feat: build AMI and OVA for RHEL 8.6 networked, nvidia and airgapped flavors

supershal opened the pull request:
**What problem does this PR solve?**:
- builds AMI and OVA for RHEL 8.6 
- fixes to install RHEL 8.6 compatible rpms of NetworkManager, nfs-utils, python3-audit 

**Which issue(s) does this PR fix?**:
<!-- Add a link to the JIRA issue for both items below
* jql=key in (D2IQ-92617)
-->
* https://d2iq.atlassian.net/browse/D2IQ-92617


**Special notes for your reviewer**:
<!--
Use this to provide any additional information to the reviewers.
This may include:
- Manual testing steps.
- Best way to review the PR.
- Where the author wants the most review attention on.
- etc.
-->


**Does this PR introduce a user-facing change?**:
<!--
If yes, add a message in the 'release-note' block below.
If this PR fixes a COPS ticket, include it after the note like: "CLI: Some bug fix. (COPS-xxxx)"
-->
```release-note
support for building AMI and OVA for RHEL 8.6
```

dkoshkin commented on ansible/roles/packages/tasks/redhat.yaml:
```diff
@@ -2,20 +2,20 @@
       - chrony
       - curl
-      - ebtables
+      - "'iptables-' if ansible_distribution_major_version == '8' }}ebtables"
```
```suggestion
      - "{{'iptables-' if ansible_distribution_major_version == '8' }}ebtables"
```

supershal replied:
fixed

dkoshkin commented on ansible/roles/packages/tasks/redhat.yaml:
```diff
@@ -2,20 +2,20 @@
   yum:
     name:
-      - audit
+      - "'python3-' if ansible_distribution_major_version == '8' }}audit"
```
```suggestion
      - "{{'python3-' if ansible_distribution_major_version == '8' }}audit"
```

supershal replied:
fixed

dkoshkin commented on ansible/roles/sysprep/tasks/redhat.yml:
```diff
@@ -60,7 +60,7 @@
       rhsm_repository:
         name: '*'
-        state: absent
+        state: disabled
```
Why is this change needed?

supershal replied:
If the user want to enable subscription at later stage, not sure if it will add the "removed" repos back. so I just opted to disable it instead of removing them.

dkoshkin commented on ansible/roles/packages/tasks/redhat.yaml:
```diff
@@ -2,20 +2,20 @@
       - sysstat
       - nfs-utils
-      - NetworkManager
+      - "NetworkManager{{ '-cloud-setup' if packer_builder_type not in ['vsphere'] }}" # OS images on cloud platforms (AWS, AZURE, GCP) supports only NetworkManager-cloud-setup package.
```
https://teamcity.mesosphere.io/buildConfiguration/ClosedSource_KonvoyImageBuilder_E2eRhel82offlineFips/3646884?buildTab=log&focusLine=3157&linesState=523.2639

>: No package matching 'NetworkManager-cloud-setup' found available, maybe something like?

```suggestion
      - "NetworkManager{{ '-cloud-setup' if packer_builder_type not in ['vsphere'] and  ansible_distribution_major_version == '8' }}" # OS images on cloud platforms (AWS, AZURE, GCP) supports only NetworkManager-cloud-setup package.
```

supershal replied:
fixed

supershal commented on ansible/roles/sysprep/tasks/redhat.yml:
```diff
@@ -70,3 +70,5 @@
     - lookup('env', 'RHSM_USER') | length > 0
     - lookup('env', 'RHSM_PASS') | length > 0
     - ansible_distribution == 'RedHat'
+    - "packer_builder_type not in ['amazon']"
```
removing redhat repositories are part of the `Remove subscription` block. This requires that system is registered with redhat and redhat repos are enabled using subscription manager. With RHEL 8.6 subscription manager throws error if repos are not subscribed before. We only need to subscribe to RHEL for vsphere non-airgapped builds only.

based on testing I will add GCP and Azure to the list.

supershal commented:
Having a lot of build failures because of `Connection to 127.0.0.1 closed.` error.

dkoshkin commented:
Something is still wrong with RHEL 8.2 offline packages
https://teamcity.mesosphere.io/buildConfiguration/ClosedSource_KonvoyImageBuilder_E2eRhel82offlineFips/3647267?buildTab=log&focusLine=2586&linesState=525.2637 :( 

```
fatal: [default]: FAILED! => {"attempts": 5, "changed": false, "failures": [], "msg": "Depsolve Error occurred: \n Problem: cannot install both NetworkManager-1:1.36.0-4.el8.x86_64 and NetworkManager-1:1.22.8-4.el8.x86_64\n  - package NetworkManager-1:1.36.0-4.el8.x86_64 obsoletes NetworkManager < 1:1.36.0-0.6 provided by NetworkManager-1:1.22.8-4.el8.x86_64\n  - package NetworkManager-cloud-setup-1:1.36.0-4.el8.x86_64 requires NetworkManager = 1:1.36.0-4.el8, but none of the providers can be installed\n  - package NetworkManager-team-1:1.22.8-4.el8.x86_64 requires NetworkManager(x86-64) = 1:1.22.8-4.el8, but none of the providers can be installed\n  - conflicting requests\n  - problem with installed package NetworkManager-team-1:1.22.8-4.el8.x86_64", "rc": 1, "results": []}
[12:59:22](https://teamcity.mesosphere.io/buildConfiguration/ClosedSource_KonvoyImageBuilder_E2eRhel82offlineFips/3647267?buildTab=log&focusLine=2587&linesState=2587)
```

supershal commented:
yum skip installing rpm only if the installed rpms's version is exactly same as whats included in the `offline` repo. 
This results in yum upgrading/downgrading the dependencies and runs into dependency conflicts with already installed packages. 

Added https://github.com/mesosphere/konvoy-image-builder/pull/531/commits/9750251c84fbbb01aa5ca20d0daa0f887321a335 to skip installing a rpm if it is already installed. 
Since an OS compatible rpm is already present on base template/OVA, we do not need to upgrade/install it. 

If we ever need to upgrade a rpm then we should use exact version in our playbooks and make sure its all dependencies are available in air-gapped bundle. 

```
[19:48:53](https://teamcity.mesosphere.io/buildConfiguration/ClosedSource_KonvoyImageBuilder_E2eRhel84offlineFips/3650465?buildTab=log&focusLine=2579&linesState=2579)
      rhel-8.4-fips: TASK [packages : install common RPMS if already not installed] *****************      rhel-8.4-fips: TASK [packages : install common RPMS if already not installed] *****************
[19:48:53](https://teamcity.mesosphere.io/buildConfiguration/ClosedSource_KonvoyImageBuilder_E2eRhel84offlineFips/3650465?buildTab=log&focusLine=2580&linesState=2580)
      rhel-8.4-fips: skipping: [default] => (item=audit)
[19:48:53](https://teamcity.mesosphere.io/buildConfiguration/ClosedSource_KonvoyImageBuilder_E2eRhel84offlineFips/3650465?buildTab=log&focusLine=2581&linesState=2581)
      rhel-8.4-fips: skipping: [default] => (item=ca-certificates)
[19:48:57](https://teamcity.mesosphere.io/buildConfiguration/ClosedSource_KonvoyImageBuilder_E2eRhel84offlineFips/3650465?buildTab=log&focusLine=2582&linesState=2582)
      rhel-8.4-fips: changed: [default] => (item=conntrack-tools)
[19:48:57](https://teamcity.mesosphere.io/buildConfiguration/ClosedSource_KonvoyImageBuilder_E2eRhel84offlineFips/3650465?buildTab=log&focusLine=2583&linesState=2583)
      rhel-8.4-fips: skipping: [default] => (item=chrony)
[19:48:57](https://teamcity.mesosphere.io/buildConfiguration/ClosedSource_KonvoyImageBuilder_E2eRhel84offlineFips/3650465?buildTab=log&focusLine=2584&linesState=2584)
      rhel-8.4-fips: skipping: [default] => (item=curl)
[19:49:02](https://teamcity.mesosphere.io/buildConfiguration/ClosedSource_KonvoyImageBuilder_E2eRhel84offlineFips/3650465?buildTab=log&focusLine=2585&linesState=2585)
      rhel-8.4-fips: changed: [default] => (item=iptables-ebtables)
[19:49:08](https://teamcity.mesosphere.io/buildConfiguration/ClosedSource_KonvoyImageBuilder_E2eRhel84offlineFips/3650465?buildTab=log&focusLine=2586&linesState=2586)
      rhel-8.4-fips: changed: [default] => (item=open-vm-tools)
[19:49:12](https://teamcity.mesosphere.io/buildConfiguration/ClosedSource_KonvoyImageBuilder_E2eRhel84offlineFips/3650465?buildTab=log&focusLine=2587&linesState=2587)
      rhel-8.4-fips: changed: [default] => (item=python3-pip)
[19:49:14](https://teamcity.mesosphere.io/buildConfiguration/ClosedSource_KonvoyImageBuilder_E2eRhel84offlineFips/3650465?buildTab=log&focusLine=2588&linesState=2588)
      rhel-8.4-fips: changed: [default] => (item=python3-netifaces)
[19:49:14](https://teamcity.mesosphere.io/buildConfiguration/ClosedSource_KonvoyImageBuilder_E2eRhel84offlineFips/3650465?buildTab=log&focusLine=2589&linesState=2589)
      rhel-8.4-fips: skipping: [default] => (item=python3-requests)
[19:49:17](https://teamcity.mesosphere.io/buildConfiguration/ClosedSource_KonvoyImageBuilder_E2eRhel84offlineFips/3650465?buildTab=log&focusLine=2590&linesState=2590)
      rhel-8.4-fips: changed: [default] => (item=socat)
[19:49:21](https://teamcity.mesosphere.io/buildConfiguration/ClosedSource_KonvoyImageBuilder_E2eRhel84offlineFips/3650465?buildTab=log&focusLine=2591&linesState=2591)
      rhel-8.4-fips: changed: [default] => (item=sysstat)
[19:49:31](https://teamcity.mesosphere.io/buildConfiguration/ClosedSource_KonvoyImageBuilder_E2eRhel84offlineFips/3650465?buildTab=log&focusLine=2592&linesState=2592)
      rhel-8.4-fips: changed: [default] => (item=nfs-utils)
[19:49:31](https://teamcity.mesosphere.io/buildConfiguration/ClosedSource_KonvoyImageBuilder_E2eRhel84offlineFips/3650465?buildTab=log&focusLine=2593&linesState=2593)
      rhel-8.4-fips: skipping: [default] => (item=NetworkManager)
```

I will file another PR in `dkp-release` to revert the changes that were made to pin certain version of the rpms

dkoshkin commented on ansible/roles/packages/tasks/redhat.yaml:
```diff
@@ -1,13 +1,27 @@
 ---
-- name: install common RPMS
+- name: find installed RPMs
```
I still don't understand why `yum` here is trying to update but that is the behavior we are seeing.

supershal replied:
yum skips updating only if the installed rpm version is exactly same as rpm available in the `offline` repo.

dkoshkin reviewed (approved):
Great job on figuring all of this out ❤️

supershal commented:
New os-packages are published for RHEL and Centos for K8s 1.24.6 version. re-triggering all the builds.

supershal commented:
> New os-packages are published for RHEL and Centos for K8s 1.24.6 version. re-triggering all the builds.
All offline e2e tests manually triggered and passed successfully.

alejandroEsc commented on make/images.mk:
```diff
@@ -476,10 +491,28 @@ rhel84-ova-offline:
 rhel84-ova-fips-offline:
 	$(MAKE) ova-rhel-8.4_offline-fips
 
+# RHEL 8.6 vSphere
```
Thank you!

alejandroEsc commented on make/images.mk:
```diff
@@ -476,10 +491,28 @@ rhel84-ova-offline:
 	$(MAKE) ova-rhel-8.4_fips
 
-.PHONY: rhel82-ova-fips-offline
+.PHONY: rhel84-ova-fips-offline
```
Nice catch

alejandroEsc reviewed (approved):
Lgtm!
